cpp/CMakeLists.txt

```diff
@@ -1023,8 +1023,10 @@ if(TARGET cudftestutil)
   set(_components_export_string COMPONENTS testing COMPONENTS_EXPORT_SET cudf-testing-exports)
 endif()
 
-install(DIRECTORY ${CUDF_SOURCE_DIR}/include/cudf ${CUDF_SOURCE_DIR}/include/cudf_test
-                  ${CUDF_SOURCE_DIR}/include/nvtext DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+install(
+  DIRECTORY ${CUDF_SOURCE_DIR}/include/cudf ${CUDF_SOURCE_DIR}/include/cudf_test
+            ${CUDF_SOURCE_DIR}/include/cudf_benchmark ${CUDF_SOURCE_DIR}/include/nvtext
```

Comment:
This seems wrong to me. Why are we including tpch-specific functions in the public libcudf.so API space?

Reply:
I want to use the datagen  (which is in `cpp/benchmarks`) from within the TPC-H examples (which is in `cpp/examples`). I understand that exposing TPC-H specific functions as part of the public API is not a good idea. I did it as more of a temporary thing and mimicked `cudf_test` until we figure out a better placement of the TPC-H examples and the datagen. But I think now is the time for it. I am open to suggestions. Maybe we can move the datagen to the examples ? Or maybe now is the time to convert the TPC-H examples into nvbenches, and move it to benchmarks ?

Reply:
I prefer converting these to benchmarks if that is ok.
Personally I would consider benchmarks and gtests also as example code and have referred people to them many times.

Reply:
I don't want to put benchmark code into libcudf.so and I don't want to require users to build the benchmarks in order to build the examples. I think we will need a new data-generator module that both benchmarks (nvbench) and examples could use. For now, I don't think we need the tpch benchmarks but I understand there is a dependency on the benchmark generator. Therefore, I would recommend moving this into benchmarks but just not using nvbench and we can resolve this as a cpp/example later.

Reply:
> I don't want to put benchmark code into libcudf.so and I don't want to require users to build the benchmarks in order to build the examples. 

I agree with this 100%. Even shipping the testing utilities with libcudf.so is a mistake we continue to pay for every day.

Reply:
As per the discussion from today, I removed the public `cudf_benchmark/tpch_datagen.hpp` header. The datagen is now completely within `cpp/benchmarks/common/cudf_tpch_datagen`.